### PR TITLE
Use Heroku's release stage

### PR DESCRIPTION
**Pull request by toobulkeh**
Heroku supports a `release` stage via Procfile which allows for custom commands to be run on deploy.

This allows us to remove repeated `db:migrate` statements from several places where our app interacts with Heroku.

Furthermore, it fixes https://github.com/thoughtbot/suspenders/issues/863 by avoiding using `bin/deploy` from CircleCI, allowing us to drop the custom CI build script from `circle.yml`.

Lastly, it paves the way for https://github.com/thoughtbot/suspenders/pull/859, by cleaning up the review app process.

### CODE REVIEW

File: spec/features/heroku_spec.rb [hunk 1]
```diff
@@ -37,16 +37,11 @@
...
 
-      expect(circle_yml).to include <<-YML.strip_heredoc
-      deployment:
-        staging:
-          branch: master
-          commands:
-            - bin/deploy staging
-      YML
+      expect(procfile).to include <<-RELEASE
+release: bundle exec rails db:migrate
```
[the remark this answers is not in the document]
Comment (toobulkeh) [reply spec/features/heroku_spec.rb]: not sure why this is worried about indentation on a single line without any whitespace. Maybe I don't even use a heredoc here?

**Comment by iwz** [comment]
@toobulkeh this looks great! didn't know Heroku has a release stage now. Mind rebasing this branch?

**Comment by toobulkeh** [comment]
@iwz done

**Comment by composerinteralia** [comment]
Closing in favor of https://github.com/thoughtbot/suspenders/pull/937

**Pull request closed without merging**